### DISCUSSION TITLE
Report Parameters

### DISCUSSION THREAD

**Discussion by rozyjaretJiv** [gerardo-lijs/CrystalReportsRunner#36] [Q&A]
Hi,
I have a requirement where I want all the parameters in a report template file . Following are the steps I want perform:
1. Create a mysql dns- **DONE**
odbchelper.createMysqlDsn(dsnName);

2. Create a report object-**DONE**
3. var report = new Report("c:/test.rpt", reportName)
{
    Connection = CrystalReportsConnectionFactory.CreateODBCSqlConnection(dsnName, databasename)
};
3. Fetch all parameters in that report for which I need to add values. I also want to fetch the datatype of each parameters . - **PENDING**
4. Add parameter values
 report.Parameters.Add(param, val);

So i want to perform the 3rd step  where I want the rpt template's parameters before adding the values in step 4.
Please help me if there is any way to achieve this using this package.
Your earliest response will be appriciated.

**Comment by gerardo-lijs** [comment]
Hi, glad to hear you are making good progress! It's possible to send all the parameters but you need to know them. If you would like to load a report file and get the parameters and datatypes, it would be something that currently is not possible. In order to do so I can imagine two options:

1) Write a GetParameters  method in the runner that load the report, read the parameters definitions and then serializes them and send them as result via named pipes to the caller. Then after getting the parameters, calling the Report method normally. This will work, but it will require to load the report twice so it might be slower. Maybe not an issue in your project.

2) Add some kind of side named pipes channel so that in the current  Report method it can provide bi-directional communication and let the caller know of the parameters. This would be harder to write, but if it works fine, it won't need to lload the report twice.

**Comment by gerardo-lijs** [comment]
Hi,

Did you make any progress on this? In case you don't necessary need 100% dynamic parameters, there is something I just remembered in which you can pass all the named-parameters and if the report has it it will be sent, otherwise just ignored. Maybe it helps in your scenario/app

**Comment by pethmani** [comment]
I got error in server database login issue  see attached screen shot and can you please help me on issue?

![image](https://github.com/user-attachments/assets/4b444a74-8c60-445c-89c8-6f70160f11e4)
